kakarot_scripts/utils/kakarot.py

```diff
@@ -140,12 +147,31 @@ def get_contract(
     return contract
 
 
+def replace_library_placeholders(
+    artifacts: Dict[str, str], deployed_libraries: List[Dict[str, str]]
+) -> Tuple[str, str]:
+    bytecode = artifacts["bytecode"]
+    bytecode_runtime = artifacts["bytecode_runtime"]
+
+    for library in deployed_libraries:
+        placeholder = f"__${library['identifier'].hex()}$__"
+        address = Web3.to_checksum_address(library["address"]).lstrip("0x")
+        bytecode = bytecode.replace(placeholder, address)
+        bytecode_runtime = bytecode_runtime.replace(placeholder, address)
+        logger.info(f"ℹ️  Replaced {library['identifier'].hex()} in bytecode")
+
+    return bytecode, bytecode_runtime
+
+
+def compute_library_identifier(library_app: str, library_name: str) -> bytes:
+    return keccak(
+        f"{SOURCE_PATH}/{library_app}/{library_name}.sol:{library_name}".encode("utf-8")
+    )[:LIBRARY_PLACEHOLDER_PREFIX_BYTES]
+
+
 async def deploy(
-    contract_app: str, contract_name: str, *args, **kwargs
+    contract: Web3Contract, *args, caller_eoa: Optional[Account] = None, **kwargs
 ) -> Web3Contract:
```

Comment:
Deploy is not returning a Web3Contract anymore but `receipt, starknet_address, evm_address`